kolibri/plugins/learn/viewsets.py

```diff
@@ -218,7 +234,7 @@ def consolidate(self, items, queryset):
                 }
             exam["missing_resource"] = any(
                 question["exercise_id"] not in available_exam_ids
-                for question in exam.get("question_sources")
+                for question in (section_questions or exam.get("question_sources"))
```

Comment:
and here:
```
for question, _ in exam_assignment_lookup(exam.get("question_sources", [])
```

This is better, because currently the `section_questions` will be collecting _all_ questions from every exam, so this will mean that if any quiz has missing resources, then all quizzes will be marked as having missing resources (because section_questions will always be all of the questions from every exam).